controllers/openstackdataplanerole_controller.go

```diff
@@ -297,18 +293,18 @@ func (r *OpenStackDataPlaneRoleReconciler) ReconcileBaremetalSet(ctx context.Con
 			dataplanev1beta1.RoleBareMetalProvisionReadyCondition,
 			condition.ErrorReason, condition.SeverityError,
 			dataplanev1beta1.RoleBaremetalProvisionErrorMessage)
-		return ctrl.Result{}, err
+		return &ctrl.Result{}, err
 	}
 
 	// Check if baremetalSet is ready
 	if !baremetalSet.IsReady() {
 		util.LogForObject(helper, "BaremetalSet not ready, Requeueing", instance)
-		return ctrl.Result{RequeueAfter: bmProvisionRetryDelay}, nil
+		return nil, nil
```

Comment:
Would not returning `nil` here would  result it moving ahead (ignored in L155-158 check) and try deploy on not yet provisioned nodes, if the deploy flag is changed to true?

Reply:
You are right, I accidentally reversed the logic between this line and line 307. I'll fix that.